2021/12/11

osmelmc opened the pull request.

osmelmc committed:
Enhance Trans (Futa) compatibility

Fix issue with Futa detection
Prevent equip strapons on futa actors
Allow Futa behave like any of both genders

osmelmc committed:
Fix Stats problems and Enhance the MCM

Fix the problem with the "TimeSpent" stats becoming negative after sex.

Allow edit some of the actor stats while SexLab is in Debug Mode. Is very basic and don't allow 0 values for tech reasons but is probably better this way. (Also probably need the real SkyUI SDK only to be compiled because I think the public one don't come the Input options)

Fix some errors on the SexLab MCM and add new enhancements .

osmelmc committed:
Update sslThreadController.psc

Fix problem with the GetAdjustPos() returning wrong value after positions bein changed

working on make the alignment hotkeys more reliable and soft

osmelmc committed:
Update sslSystemConfig.psc

Set the LipSync and the Expressions enabled by default

Fix the problem with the creature animations don't bein automatically registered after import the configuration with the creatures allowed.

osmelmc committed:
Merge branch 'eeveelo:development' into My-Branch

osmelmc committed:
Update sslSystemAlias.psc

osmelmc committed:
Update README.md

Update the Change Logs
between

osmelmc committed:
Fix problem with the RealTime native function on AE

Users on AE are reporting issues with the Orgasm and the automatic transition between animations stages. Ed86 do few test and seems like the GetCurrentGameRealTime function is not working on AE.

I set again my version of the function that is working fine on SE, that way you can use it as reference if the Native is not working for real.

Remember I can't test AE right now, so I can't know the real reason!

osmelmc committed:
Merge branch 'My-Branch' of https://github.com/osmelmc/SexLab into My-Branch

eeveelo reviewed (changes requested):
Allowing GetCurrentGameRealTime() to use GetCurrentRealTime() in it's place would result in differently formatted timestamps possibly being used in parts if the user to for some stupid reason change their timescale mid scene, which could lead to infinite/very long loops in some cases. It's highly unlikely to ever happen though, so maybe a good solution to the 0 timescale problem anyways.

In any case though, I'll make the change to the dll's native function before next release, so remove the non-native version from this pull-request and I'll update the native function to the same.

eeveelo commented:
Actually, possibly bad idea to default it to GetCurrentRealTime(). Since RecordThread() is changed to be using the new function via RealTime[0], it means it'd be possible for differently formatted timestamps to be used in the stat tracking, resulting in errors.

osmelmc commented:
> 0

Is bad idea but is not the worse. the only issue is the value of the last sex time, but nobody seems to be using it. 

the previous time problem was caused by the difference between the formats of the RealTime and StartedAt parameters on the RecordThread function.

in this case should not happen because both will have the same format at less the TimeScale be settled to 0 in meddle of the animation but even in that case the users now the users have a way to deal with that.

use the GetCurrentRealTime() as alternative for the TimeScale=0 is the best solution I know and will keep the function running on the worse of the cases. 

By the way I'm still don't know for what reason the native function is not working on AE at least for some users. So I will let you my function here for reference, and you can remove it once the native function be fixed.

osmelmc committed:
Fix Anim Sync issues

Fix the Chaurus problem on the first stage
Help to fix the Sync problem usually on the first animation stage when one of the actors seems to be ignored, leading to animations not longer advancing. (The problem is bigger than that and this is just a temporal solution)

osmelmc committed:
Update sslSystemAlias.psc

the "debug_Cleanup()" function seems to be causing CTD in some PCs. I don't know the reason so I just allow it while sexlab is in debug mode.

osmelmc committed:
Include the BedOffsets on the SexLab Animation extorted/imported to JSON fies

osmelmc committed:
Revert "Fix problem with the RealTime native function on AE"

This reverts commit dca3649886ad5a504d8103b86ecd9883b55cb90a.
